crates/sui-graphql-rpc/src/types/transaction_block_kind.rs

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use super::{big_int::BigInt, date_time::DateTime, epoch::Epoch, object::Object};
+use crate::context_data::db_data_provider::PgManager;
+use async_graphql::{ComplexObject, Context, Result, ResultExt, SimpleObject, Union};
+
+#[derive(Union, PartialEq, Clone, Eq)]
+pub(crate) enum TransactionBlockKind {
+    ConsensusCommitPrologueTransaction(ConsensusCommitPrologueTransaction),
+    GenesisTransaction(GenesisTransaction),
+    ChangeEpochTransaction(ChangeEpochTransaction),
+    // ProgrammableTransactionBlock(ProgrammableTransactionBlock),
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, SimpleObject)]
+#[graphql(complex)]
+pub(crate) struct ConsensusCommitPrologueTransaction {
+    #[graphql(skip)]
+    pub(crate) epoch_id: u64,
+    // # TODO: This is the "leader round" -- does this line up with
+    // # checkpoints? In which case, it may suffice to have a `Checkpoint`
+    // # here.
+    pub(crate) round: Option<u64>,
```

Comment:
No this is a consensus round and can and will reset on epoch changes

Reply:
Got it, thank you!